### PR TITLE
Implement GaussianSkewtScatterSelector to emulate COSMOS-like photometric redshifts

### PR DESCRIPTION
**Pull request by jtmyles**
This PR implements an example selector to emulate COSMOS photometric redshifts with biases relative to the truth. The biased values are drawn from a model with parameters stored in a csv file. 

This linked [example notebook](https://gist.github.com/jtmyles/b16e0aa5c8f89c48248fadf73aa22335) shows the selector using model parameters fit on zCOSMOS data. The mock photo-z residuals mostly reproduce the residuals seen in the test dataset. 

The model implemented contains an inner Gaussian core distribution from Yin+2025 and a broader tail distribution set by a skewed t-distribution. Samples from the model are re-drawn up to three times for bias values that yield photo-z outside the bounds [0,6]. 

This model is trained on a sample of biased spectroscopic data, so cannot be expected to represent the full variance in the overall population. Better motivated alternatives to this simplistic two component model could reproduce the observed data distribution better, but since this data distribution is expected to underestimate the actual variance of the population, improving the model without appropriately inflating the model variance may simply amount to more overfitting to the biased spectroscopic data. The small test set used for the default model has fewer than 25 spectra fainter than i > 23, so the default model is especially limited at those magnitudes. 

The csv file stores component parameters for a given i-band magnitude and z bin. The cosmos_selector.py file re-organizes this into a dictionary for sampling. If anyone has suggestions on improving the way the data is stored and loaded please let me know.

### CODE REVIEW

**Review by sschmidt23 — CHANGES_REQUESTED** [review]
This is mostly exactly what we want, however, reading parameters from a separate csv can lead to problems in keeping track of parameters when we update pipelines, as the csv can get out of sync with params in the yaml files.  I think we should change this to either 1) be a dictionary of parameters that is specified by the user and has a default that lives in the code, or 2) have a stage that creates a dictionary pickle file.  Option 1 sounds better to me, there are small examples like e.g. regression_params in rail_flexzboost:
https://github.com/LSSTDESC/rail_flexzboost/blob/c60ff8462e55f15aa4507b01460eb192081c3bda/src/rail/estimation/algos/flexzboost.py#L107
Though, since this dictionary will be a bit more extensive, I'd just define a dictionary at the top of the file just outside of the `class COSMOSSelector(Selector)` definition that has the default COSMOS dictionary values, i.e.
default_selector_dict=dict(gaussian_core=[array of stuff], skey_student_t_tail=[array of stuff], etc...])
and then replace `col_name` with `selector_dict` and give its the default value of `default_selector_dict`.

If we do it this way, then a user could specify a different input dictionary and use this for another sample, e.g. the PAUS narrow band survey mentioned recently.  That also makes me wonder about the name `COSMOSSelector`, maybe we should change to `GaussianLognormalScatterSelector` or something more generic like that to reflect that you can change behavior based on the input dictionary?

**Comment by jtmyles** [comment]
Thanks for these comments, Sam. This makes sense. I'll get started on implementing Option 1. And I'll rename the selector to `GaussianSkewtdistScatterSelector` or something similar.

**Review by sschmidt23 — APPROVED** [review]
Looks good, scatter looks more like what I think we expect, thanks Justin
